eng/pipelines/ci.yml

```diff
@@ -334,10 +334,83 @@ jobs:
         artifact: CoverageReport
 
       - script: |
-          npm pack
+          pnpm pack
           AbsolutePkgPath=$(find "$(pwd)" -name "autorest-python-*" -maxdepth 1)
-          cd ../../..
-          npm install $AbsolutePkgPath
+          cd ../..
+          pnpm install $AbsolutePkgPath --verbose
         displayName: Test tarball
         workingDirectory: $(Build.SourcesDirectory)/autorest.python/packages/cadl-python/
 
+  - job: "Python_Generator_Core_CI"
+    displayName: "Run Python Generator Core CI Checks"
+
+    strategy:
+      matrix:
+        Linux_Python37:
+          OSName: "Linux"
+          OSVmImage: "MMSUbuntu20.04"
+          PythonVersion: "3.7"
+        Linux_Python310:
+          OSName: "Linux"
+          OSVmImage: "MMSUbuntu20.04"
+          PythonVersion: "3.10"
+
+    pool:
+      name: azsdk-pool-mms-ubuntu-2004-general
+      vmImage: $(OSVmImage)
+
+    steps:
+      - checkout: self
+      - checkout: azure-sdk-tools
+      - template: eng/common/testproxy/test-proxy-tool.yml@azure-sdk-tools
+        parameters:
+          TestProxy: true
+          templateRoot: azure-sdk-tools
+      - task: NodeTool@0
+        displayName: "Install Node.js $(NodeVersion)"
+        inputs:
+          versionSpec: "$(NodeVersion)"
+
+      - task: UsePythonVersion@0
+        displayName: "Use Python 3.7"
+        inputs:
+          versionSpec: 3.7
+
+      - script: |
```

Comment:
I would recommend seperating those commands in independent steps, makes it much easier to see what is wrong, would create a reusable template if you are doing that in multiple paces.

Reply:
ah sweet, let me look into the reusable templates. Thanks!